packages/use-presence/src/index.tsx

```diff
@@ -107,3 +109,57 @@ export default function usePresence(
     isExiting
   };
 }
+
+export type UseUniqueDataPresenceOptions<T> = {
+  /** Check if two instances of type T are equal, defaults to a === comparison */
+  equalityCheck?(a: T, b: T): boolean;
+  /** Check if an instance of type T is considered valid to render, defaults to Boolean function */
+  validationCheck?(a: T): boolean;
+} & UsePresenceOptions;
+
+function defaultEqualityCheck<T>(a: T, b: T) {
+  return a === b;
+}
+function defaultValidationCheck<T>(a: T) {
+  return Boolean(a);
+}
+
+export function useUniqueDataPresence<T>(
```

Comment:
Bike-shedding a bit on the name here:

- `usePresenceSwitch`
- `useSwitchPresence`
- `useItemPresence`

I kind of like the term "switch", since it expresses the concept that there can be multiple options but only one active at a time. `react-router` also uses the term.

I think my preference would be `usePresenceSwitch`—what do you think?

Reply:
I also like `usePresenceSwitch` though I also went with updating `data` & `dataInput` to `item` & `latestItem`. So maybe now you might prefer `useItemPresence` but I think both names together are still self explanatory, but let me know how you feel about it now 👍